samtranslator/model/sam_resources.py

```diff
@@ -2407,3 +2418,148 @@ def _set_resolver_code_settings(resolver_code_settings: aws_serverless_graphqlap
         # TODO: Add FileNamePatterns defaults once implemented
         resolver_code_settings.FunctionsFolder = resolver_code_settings.FunctionsFolder or "functions"
         resolver_code_settings.ResolversFolder = resolver_code_settings.ResolversFolder or "resolvers"
+
+    def _construct_appsync_function_configurations(
+        self,
+        functions: Dict[str, aws_serverless_graphqlapi.Function],
+        datasources: Optional[Dict[str, aws_serverless_graphqlapi.DynamoDBDataSource]],
+        api_id: Intrinsicable[str],
+    ) -> List[Resource]:
+        resources: List[Resource] = []
+
+        none_datasource = self._check_and_construct_none_datasource(functions, api_id)
+
+        if none_datasource:
+            resources.append(none_datasource)
+
+        for relative_id, function in functions.items():
+            func_config = FunctionConfiguration(
+                logical_id=self.logical_id + relative_id,
+                depends_on=self.depends_on,
+                attributes=self.resource_attributes,
+            )
+
+            # "Id" is a mutually exclusive property to every other property. If this function has it
+            # defined, then we make sure it's the only property, and continue to next function.
+            if function.Id:
+                keys = remove_none_items(function.dict()).keys()  # remove undefined properties, then get keys
+                if len(keys) != 1:
+                    raise InvalidResourceException(
+                        relative_id, "'Id' cannot be defined with other properties in Function."
+                    )
+                continue
+
+            if not function.InlineCode and not function.CodeUri:
```

Comment:
I think I'll make a utility function that generates these mutually exclusive value errors. It could just take two values (maybe even a list and I can support multiple x amount of values) and a relative ID and throw these errors since the message is pretty consistent regardless of the property

Reply:
I'll make this in a follow up PR cause I really want to get this change out and this affects other properties in `GraphQLApi`, not just `Function`.

Reply:
Adding a TODO here might help remember